lib/charms/observability_libs/v0/metrics_endpoint_discovery.py

```diff
@@ -1,27 +1,41 @@
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-"""TODO: Add a proper docstring here.
+"""# MetricsEndpointDiscovery Library.
```

Comment:
Testing instructions seems to contain some copy/paste artifacts. sweat_smile To get it to work I had to replace hooks/discovery with hooks/metrics_endpoint_change. 😅 

Is there a way we could change this to not have to extend self.on manually, for instance through a MetricsEndpointDiscovery class we could subscribe to in a fashion similar to [what is being done here](https://github.com/canonical/prometheus-k8s-operator/blob/8b8e23b929e27d6ffc909c79dc480cb39726c760/src/charm.py#L90)?

Reply:
I too think that this would benefit from the constructor-centric approach we use in the other O11y libs.

Reply:
I believe I've addressed everything.